docs/specification.md

```diff
@@ -65,14 +80,63 @@ The owner of the Orchestrator can then collect the fees using `orchestrator.coll
 ![CollectBlend](./flowcharts/Orchestrator/collectBlend.png)
 
 ## Contracts
+
+### Multisig
+The contract is non-upgradeable and does not have a single owner.
+
+Storage:
+* `uint public nonce`
+* `address[] public owners`
+* `mapping(address => bool) public isOwner`
+* `uint public threshold`
+
+Methods:
+1. `execute(address destination, uint256 value, bytes memory data, bytes[] memory signatures)`
+   - **Behavior:** If all the signatures are correct, and if the number of signatures is more than threshold, sends a transaction to `destination` transferring `value` tokens nad providing `data` as the transaction data. Otherwise, fails. Note that the sgnatures must be provided in order (by signer's address, ascending).
+      - Data to sign:
+         - `0x1900`,
+         - Multisig address,
+         - `destination`,
+         - `value`,
+         - `data`,
+         - `nonce`.
+      - The nonce is increased by 1 after this operation.
+      - Reverts if `newThreshold` is more than `newOwners.length`.
+   - **Access:** anyone
+   - **Exceptions:**
+      - Threshold not met
+      - Invalid signature
+      - The addresses must be provided in the ascending order
+      - The supplied threshold is more than the number of owners
+
+2. `rotateKeys(address[] memory newOwners, uint newThreshold, bytes[] memory signatures)`
+   - **Behavior:** If all the signatures are correct, and if the number of signatures is more than threshold, sets the new set of eligible signers and the new threshold. Note that the sgnatures must be provided in order (by signer's address, ascending).
+      - Data to sign:
+         - `0x1900`,
+         - Multisig address,
+         - `newOwners`,
+         - `newThreshold`,
+         - `nonce`
+      - Deletes old owners from the `isOwner` mapping.
+      - Inserts the new owners to the `isOwner` mapping.
+      - Replaces `owners` with the new owners list.
+      - Updates `treshold` with the new threshold.
+      - The nonce is increased by 1 after this operation.
+   - **Access:** anyone
+   - **Exceptions:**
+      - Threshold not met
+      - Invalid signature
+      - The addresses must be provided in the ascending order
+
 ### Blend Token
 The contract is an upgradeable ERC-20 token with custom `transfer` and `transferFrom` logic.
 
 The contract has an owner. The owner can be updated via a two-step updating approach (`transferOwnership` + `acceptOwnership`).
 
 Storage:
-* `public address orhestrator`
-* `public address registry`
+* `bool public distributionPhase`
+* `Registry public registry`
```

Comment:
Maybe `address public registry`?

Reply:
We need its interface to access the methods, so it has to be `Registry`. If we just checked it against the reference address (like in ownership checks), it would be `address`.